src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -211,28 +231,29 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, workspaceDAO:
 
   def getACL(workspaceName: WorkspaceName): PerRequestMessage =
     dataSource inTransaction { txn =>
-      withWorkspaceContextAndPermissions(workspaceName, GCSAccessLevel.Owner, txn) { workspaceContext =>
-        Try(gcsDAO.getACL(userInfo.userId, workspaceContext.bucketName)) match {
+      withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevel.Owner, txn) { workspaceContext =>
+        Try(gcsDAO.getACL(workspaceContext.bucketName, workspaceName)) match {
           case Success(acl) => RequestComplete(StatusCodes.OK, acl)
           case Failure(err) => RequestComplete(StatusCodes.Forbidden, s"Can't retrieve ACL for workspace ${workspaceName}: " + err.getMessage())
         }
       }
     }
 
-  def putACL(workspaceName: WorkspaceName, acl: String): PerRequestMessage =
+  def updateACL(workspaceName: WorkspaceName, aclUpdates: Seq[WorkspaceACLUpdate]): PerRequestMessage =
     dataSource inTransaction { txn =>
-      withWorkspaceContextAndPermissions(workspaceName, GCSAccessLevel.Owner, txn) { workspaceContext =>
-        Try(gcsDAO.putACL(userInfo.userId, workspaceContext.bucketName, acl)) match {
-          case Failure(err) => RequestComplete(StatusCodes.Forbidden, s"Can't set ACL for workspace ${workspaceName}: " + err.getMessage())
-          case _ => RequestComplete(StatusCodes.OK)
+      withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevel.Owner, txn) { workspaceContext =>
+        val updateErrors = gcsDAO.updateACL(workspaceContext.bucketName, workspaceName, aclUpdates)
+        updateErrors.size match {
```

Comment:
style point: I were prefer to use if (updateErrors.isEmpty)... it expresses your intent better and isEmpty may be optimized.